include/NAS2D/Renderer/RendererOpenGL.h

```diff
@@ -92,10 +92,14 @@ class RendererOpenGL: public Renderer
 
 	void initGL();
 	void initVideo(unsigned int resX, unsigned int resY, bool fullscreen, bool vsync);
+	std::string getGlString(unsigned int stringId) const;
+	std::string getGlRendererString() const;
+	std::string getGlVendorString() const;
+	std::string getGlVersionString() const;
+	std::string getGlShadingLanguageVersionString() const;
```

Comment:
I suppose that's one way to get around adding an OpenGL include to the header file.

----

The [`glGetString`](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glGetString.xhtml) function takes a `GLenum` parameter. Here an `unsigned int` is being passed. That could potentially lead to a skew in types between the caller and the API.

Reply:
`GLenum` is a typedef for an `unsigned int`, additionally, each string ID is a `#define` no bigger than `#FFFF`.

And yes, I did not want to pollute the header with an include for OpenGL. :)

Reply:
The first part of my comment was aimed more at the specialized functions for each enum value. By not taking a parameter, they don't have a type dependence on the enum.

I only realized a bit later that `getGlString` was a general method used by all of them. That sort of spoils the attempt in a way, since it does take an enum value.

I would strongly recommend against changing types just to prevent namespace pollution in header files. Particularly since types and type aliases sometimes change overtime, like during the transition from 32-bit to 64-bit. You might be able to get away with it here, but I recommend that you don't.

----

Another thought is those methods don't need access to class member variables, so they could potentially be made static. Of course, if you have a static method, you can potentially convert it to a regular function in an unnamed namespace in the source file, and completely remove all traces of it from the header file.

A benefit to that approach, is you can use types in the signature that are only included by the implementation file, so that's another approach to avoid namespace pollution in the header file.

That was the approach I took in #547.

Reply:
Yeah, I like your approach a little better. It's just icky that you have to use the value provided by OpenGL.

My approach was since there are so few valid enum values in the first place, using a common function to call in to allowed for one point of failure if either an invalid value was passed or more values were added, in that case, just add another function.

Could even go so far as to have a private `ConvertOpenGLStringToEnum` function that did the conversion via `switch` instead of just a direct call.

Reply:
You mean the reverse operation? I'm afraid I don't quite follow that last part.

Reply:
It's an alternative way to remove the header dependency by only using "our" types.

Reply:
Ahh, I assume you mean replace their enum with our own enum, and then translate between them with a `switch`.

Reply:
Yes. :)